### PR TITLE
Multiple keyboard layout support

### PR DESCRIPTION
**Pull request by RandyTheDev**
Added the ability to automatically detect a user's keyboard layout. If a player uses a non-QWERTY keyboard layout, the plugin will intelligently adjust which keys it lights depending on the layout.

### New Features
- Supports any keyboard layout.
- The Quickload button will now only be illuminated if you have quick-loading enabled.

### Other enhancements
- The KSP Logo lighting scheme has been tuned for Logitech Keyboard layouts.
- The keyboard lighting effects will no longer disappear while waiting on loading screens.

### Housekeeping
- Extensive refactor
- Compiled against the KSP 1.4 API.
- Uses a newer version of the Logitech SDK.
- There are some basic unit tests.

## Next Steps
- Bringing this version back up to feature parity with KSP-Chroma, then going beyond.

### CODE REVIEW

**Comment by RandyTheDev** [comment]
Hey don't merge this yet, Logitech Developer support got in contact with me and I think I have a better way of doing multi-layout support 😁

**Comment by RandyTheDev** [comment]
So Logitech's approach was a dead end... On the plus side, I implemented dynamic loading of the library, so KSP can now find the x64 or x86 version of the library depending on the runtime. The libraries should be installed as follows:

- KSP
  - GameData
    - KSPLogiRGB
      - Logitech
        - x86
          - LogitechLedEnginesWrapper.dll
        - x64
          - LogitechLedEnginesWrapper.dll

**Comment by RandyTheDev** [comment]
I also still have an idea on how to do multi layouts properly.

**Comment by battlemoose** [comment]
@RandyTheDev This looks great! You've obviously put in a lot of effort. Are you ready for me to merge at this point?

**Comment by RandyTheDev** [comment]
Hey @battlemoose!

I believe I'm ready now. I believe the layout handling code is as good as KSP will allow it to be! Please be aware of the commit where I licensed my individual contributions under the Apache license, which is GPL compatible. Commit: ada45f7

Information on GPL compatibility: https://www.gnu.org/licenses/license-list.html#apache2

**Comment by RandyTheDev** [comment]
Also, make sure you download the latest version of the Logitech APIs, so that the release zip has a structure copying the one described above.

File: KSP-LogiRGB/KSP-LogiRGB/Tests/LayoutProviderTests.cs [hunk 1]
```diff
@@ -183,7 +186,7 @@ public void TestDvorak()
                 AssertMapped(KeyCode.R, KeyCode.O);
                 AssertMapped(KeyCode.L, KeyCode.P);
                 AssertMapped(KeyCode.Slash, KeyCode.LeftBracket);
-                AssertMapped(KeyCode.Plus, KeyCode.RightBracket);
```
Comment (RandyTheDev) [inline KSP-LogiRGB/KSP-LogiRGB/Tests/LayoutProviderTests.cs]: While the Dvorak keycode is a plus symbol, in KSP, it is registered as a `KeyCode.Equals`

**Pull request merged**